pkg/services/ngalert/notifier/redis_peer.go

```diff
@@ -273,6 +270,7 @@ func (p *redisPeer) membersSync() {
 	// those out that have failed to send a heartbeat during the heartbeatTimeout.
 	peers := p.filterUnhealthyMembers(members, values.Val())
 	sort.Strings(peers)
+	peers = slices.Compact(peers)
```

Comment:
Why would we need this? In case the keys changes while iterating over it? We should add a comment about it.

Reply:
SCAN might return duplicate elements ([docs](https://redis.io/commands/scan/)): 

> However because SCAN has very little state associated (just the cursor) it has the following drawbacks:
> 
> A given element may be returned multiple times. It is up to the application to handle the case of duplicated elements, for example only using the returned elements in order to perform operations that are safe when re-applied multiple times.

Added the Compact after sorting to filter out duplicate entries that could cause inconsistencies when calculating the replicas' position. 

I'll add a comment for this

Reply:
Updated in [6c8c684](https://github.com/grafana/grafana/pull/70607/commits/6c8c6843bc435940953d15ac3c5c67ae140084c7)

Let me know if this looks ok

Reply:
Nice, let's go!